### PR TITLE
Test: Convert rolling upgrade test to have task per wire compat version

### PR DESCRIPTION
**Pull request by rjernst**
This commit changes the rolling upgrade test to create a set of rest
test tasks per wire compat version. The most recent wire compat version
is always tested with the `integTest` task, and all versions can be
tested with `bwcTest`.

### CODE REVIEW

**Comment by rjernst** [comment]
I have a version of this PR running on 5.x as well. FWIW, the time to run the `bwcTest` task there is about 9 minutes on my laptop.

File: build.gradle [hunk 1]
```diff
@@ -76,26 +77,27 @@ for (String line : versionLines) {
...
     if (major == prevMajor && minor > lastPrevMinor) {
       prevMinorIndex = versions.size() - 1
       lastPrevMinor = minor
     }
   }
 }
-if (versions.toSorted() != versions) {
-  throw new GradleException('Versions.java contains out of order version constants')
+if (versions.toSorted { it.id } != versions) {
+  println "Versions: ${versions}"
```
Comment (s1monw) [inline build.gradle]: leftover?

File: build.gradle [hunk 2]
```diff
@@ -167,6 +168,9 @@ subprojects {
     "org.elasticsearch.plugin:parent-join-client:${version}": ':modules:parent-join',
     "org.elasticsearch.plugin:percolator-client:${version}": ':modules:percolator',
   ]
+  if (wireCompatVersions[-1].snapshot) {
```
Comment (s1monw) [inline build.gradle]: can you leave a comment what this is doing? it helps me to understand this better

File: qa/rolling-upgrade/build.gradle [hunk 3]
```diff
@@ -18,71 +18,89 @@
  */
 
 import org.elasticsearch.gradle.test.RestIntegTestTask
+import org.elasticsearch.gradle.Version
 
 apply plugin: 'elasticsearch.standalone-test'
 
-task oldClusterTest(type: RestIntegTestTask) {
-  mustRunAfter(precommit)
+// dummy task for all bwc tests
```
Comment (s1monw) [inline qa/rolling-upgrade/build.gradle]: can you make this commetn more verbose why we do this and what the implications are?

File: qa/rolling-upgrade/build.gradle [hunk 4]
```diff
@@ -18,71 +18,89 @@
...
+task bwcTest {
```
Comment (s1monw) [inline qa/rolling-upgrade/build.gradle]: should we document what `gradle bwcTest` does in the cheat sheet?

**Review by s1monw — APPROVED** [review]
left some comments LGTM in general

File: build.gradle (hunk 1 above)
Comment (rjernst) [inline build.gradle]: No, this is meant to print the versions before failing if we find the versions in Version.java are not in sorted order.

**Comment by rjernst** [comment]
@s1monw I added a section in TESTING.asciidoc and improved the comments. Does that help?

**Review by s1monw — APPROVED** [review]
LGTM thanks

**Pull request merged**